sharktank/setup.py

```diff
@@ -14,15 +14,22 @@
 
 # Setup and get version information.
 VERSION_INFO_FILE = os.path.join(SETUPPY_DIR, "version_info.json")
+VERSION_INFO_RC_FILE = os.path.join(SETUPPY_DIR, "version_info_rc.json")
 
 
-def load_version_info():
-    with open(VERSION_INFO_FILE, "rt") as f:
+def load_version_info(version_file):
+    with open(version_file, "rt") as f:
         return json.load(f)
 
 
-version_info = load_version_info()
-PACKAGE_VERSION = version_info["package-version"]
+try:
+    version_info = load_version_info(VERSION_INFO_RC_FILE)
+except FileNotFoundError:
+    print("version_info_rc.json not found. Default to dev build")
+    version_info = load_version_info(VERSION_INFO_FILE)
```

Comment:
Oh, interesting choice to have two different files. Did you consider other approaches? I can think of a few:
* Overwrite the file (makes git index dirty for development though)
* Have `setup.py` load from `version_info.json` and do the `PACKAGE_RC_VERSION = PACKAGE_BASE_VERSION + "rc" + datetime.today().strftime("%Y%m%d")` here

Multiple ways to do it, and what you have works.

Reply:
I think there are even more:

* Read the version from a `version_info` (text) and override with the updated one.
  * Use the version info file in the `pyproject.toml` making the `setup.py` obsolete:
    `version = {file = version_info}`
  * Drawback: Git gets dirty, limited flexibility.

With regards to the ones you listed:

* I would like to avoid that the index gets dirty.
* This introduces a dependency on `packaging` which isn't strictly needed. Furthermore, with this every builder calculates the version instead of computing this _once_ in the CI.

But I am happy to iterate in a follow up but this one should also be compatible with what we will need for IREE.